File: POO_PHP/cardDeck.php
<?php

require_once('randomizable.php');

class CardDeck implements Randomizable {
    private $colors;
    private $values;

    public function __construct($colors, $values) {
        $this->colors = $colors;
        $this->values = $values;
    }

    public function roll() {
        $color = rand(1, count($this->colors));
        $value = rand(1, count($this->values));
        return "Color: $color, Value: $value";
    }
}
?>

File: POO_PYTHON/CardDeck.py
import random

class CardDeck:
    def __init__(self, colors, values):
        self.colors = colors
        self.values = values

    def roll(self):
        color = random.randint(1, len(self.colors))
        value = random.randint(1, len(self.values))
        return color, value
